spark/src/main/java/io/ddf/spark/content/SchemaHandler.java

```diff
@@ -106,5 +127,260 @@ public static Schema getSchemaFromDataFrame(DataFrame rdd) {
+   *
+   * @param row                  which hold multiple columns to convert
+   * @param listConvertColumns   given columnIndex & columnName &  type to convert to
+   * @param statisticAccumulator statistic
+   */
+  private Boolean convertRowType(Row row, List<Tuple3<Integer, String, Schema.ColumnType>> listConvertColumns, Accumulator<ApplySchemaStatistic> statisticAccumulator) {
+    boolean result = true;
+    statisticAccumulator.localValue().increaseLineProcessed();
+    for (Tuple3<Integer, String, Schema.ColumnType> tuple3 : listConvertColumns) {
+      if (!isConvertible(row, tuple3._1(), tuple3._3())) {
+        statisticAccumulator.localValue().addColumnFailed(tuple3._2(), row.getString(tuple3._1()));
+        result = false;
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Check whether we could convert data at row[index] requiredType @see Schema.ColumnType
+   *
+   * @param row:          Which row to convert
+   * @param index:        index of column in row to convert
+   * @param requiredType: type we need to convert to
+   * @return whether we could convert row[index] to requiredType
+   */
+  private boolean isConvertible(Row row, Integer index, Schema.ColumnType requiredType) {
+
+    try {
+      if(row.get(index) == null) return false;
+      String columnData = row.get(index).toString().trim();
+      if (columnData == null) return false;
+      switch (requiredType) {
+        case TINYINT:
+          return Byte.valueOf(columnData) != null;
+        case SMALLINT:
+          return Short.valueOf(columnData) != null;
+        case INT:
+          return Integer.valueOf(columnData) != null;
+        case BIGINT:
+          return Long.valueOf(columnData) != null;
+        case FLOAT:
+          return Float.valueOf(columnData) != null;
+        case DOUBLE:
+          return Double.valueOf(columnData) != null;
+        case DECIMAL:
+          return new BigDecimal(columnData) != null;
+        case STRING:
+          return true;
+        case BINARY:
+          return columnData.getBytes() != null;
+        case BOOLEAN:
+          return Boolean.valueOf(columnData) != null;
+        case TIMESTAMP:
+          return Timestamp.valueOf(columnData) != null;
+        case DATE:
+          return Date.valueOf(columnData) != null;
+        case ARRAY:
+        case ANY:
+        case BLOB:
+          return true;
+        case STRUCT:
+        case MAP:
+          return false; //need more info to parse
+      }
+    } catch (ClassCastException | NullPointerException | IllegalArgumentException nfe) {
+      mLog.debug(nfe.getMessage());
+    }
+    return false;
+  }
+
+  /**
+   * get list column need to be convert
+   *
+   * @return List<Tuple3<Integer,String,ColumnType>> = ColumnIndex,ColumnName,ColumnType
+   */
+  protected List<Tuple3<Integer, String, Schema.ColumnType>> getListConvertColumn(Schema fromSchema, Schema toSchema) throws DDFException {
```

Comment:
Hi @Huandao0812 ,
I don't know how to do in DataFrame, let I check. 
Currently my approach is using Rdd<Row> instead.
Do you know whether DataFrame could give out the statistic ?